### PR TITLE
KAFKA-2340; improve KafkaConsumer Fetcher test coverage

### PR DESCRIPTION
**Pull request by hachikuji**

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/clients/MockClient.java [hunk 1]
```diff
@@ -180,4 +213,13 @@ public Node leastLoadedNode(long now) {
         return this.node;
     }
 
+    /**
+     * Request predicates give the tester the ability to set an expectation on the request that
```
Comment (guozhangwang) [inline clients/src/test/java/org/apache/kafka/clients/MockClient.java]: This comment is not very intuitive to understand: how about: ".. set an expectation on the request that this response is going to be returned for..."

Comment (hachikuji) [inline clients/src/test/java/org/apache/kafka/clients/MockClient.java]: Yeah, that might be better. Do you think the name RequestPredicate makes the intended usage clear? I was also thinking of calling it RequestMatcher or RequestAssertion.

Comment (guozhangwang) [inline clients/src/test/java/org/apache/kafka/clients/MockClient.java]: Maybe RequestMatcher?

**Comment by hachikuji** [comment]
@guozhangwang Tried to make RequestMatcher comment clearer. Let me know if it helps.

**Comment by guozhangwang** [comment]
Thanks for the patch. LGTM.

**Pull request closed without merging**